src/Schema/TableDiff.php

```diff
@@ -44,6 +44,17 @@ public function __construct(
         private readonly array $modifiedForeignKeys = [],
         private readonly array $droppedForeignKeys = [],
     ) {
+        if (count($this->modifiedForeignKeys) < 1) {
```

Comment:
That's a convoluted way to check for `0`, or am I missing something?

Reply:
Why would you want to silently return for negative values? Semantically, I'd lean towards `=== []` , we want to detect if the value differs from the default value (regardless of what it is), right? I mean, we want to deprecate this parameter, don't we?

Reply:
> Semantically, I'd lean towards `=== []` ,

This is a PHP-ism. See https://github.com/doctrine/dbal/pull/6621#discussion_r1855528698 for more details. Let's do `count($array) === 0` then.

Reply:
Works for me